### DISCUSSION TITLE
Have you ever experienced slow page loading times when running rod in GitHub Actions?

### DISCUSSION THREAD

**Discussion by cecobask** [go-rod/rod#1091] [Q&A]
I'm using GitHub Actions to automate web scraping activities on the IMDb website. When running Rod in a GitHub Actions workflow (GitHub-hosted runner), individual pages sometimes wait 3-4 minutes for the [Page.WaitLoad](https://pkg.go.dev/github.com/go-rod/rod#Page.WaitLoad) function. On my device, the same activity finishes quite quickly - less than 10 seconds.
Internet speeds vary and GitHub runners have fewer computing resources than my device, but it seems like a huge time difference. This in turn lets the GitHub Actions workflow run too long and ends up timing out most of the time.

This is an example: https://github.com/cecobask/imdb-trakt-sync/actions/runs/9956730607/job/27507259798

Does anyone have suggestions on how to remediate such issues?

**Comment by ysmood** [comment]
You don't have to use load event if you use MustElement or any other element based query.

Page load event is very complex and not that reliable.
